### PR TITLE
Add forward force parameter to Impedance controller

### PR DESCRIPTION
**Pull request by domire8**
## Description

- #13 

This PR solves the issue by adding a parameter `forward_force` to the base impedance controller. This parameter (bool) controls if the error in wrench/torque should be added to the control output of the controller.

In order to finish my tasks of this iteration, I guess I need to open this as well. I had it ready since last week but I don't like the PR hence I procrastinated :smile:. First, I don't like the name `forward_force` so much because it should be wrench for the Cartesian space and torque for the joint space. Second, I wasn't sure if this should be completely up to the user to decide of if we do it by default in the compliant twist controller.

## Review guidelines

Estimated Time of Review: 5 minutes

### CODE REVIEW

**Comment by domire8** [comment]
Rebased after push on develop

File: source/controllers/include/controllers/impedance/Impedance.hpp [hunk 1]
```diff
@@ -64,6 +64,8 @@ class Impedance : public IController<S> {
       damping_; ///< damping matrix of the controller associated to velocity
   std::shared_ptr<state_representation::Parameter<Eigen::MatrixXd>>
       inertia_; ///< inertia matrix of the controller associated to acceleration
+  std::shared_ptr<state_representation::Parameter<bool>>
+      forward_force_; ///< flag to decide of force error should be passed on
```
Comment (eeberhard) [inline source/controllers/include/controllers/impedance/Impedance.hpp:68]: nitpick: I think I would prefer the more explicit and verbose name for the parameter. Also there is a typo in the comment
```suggestion
      feed_forward_force_; ///< flag to decide if force error should be passed on
```

File: source/controllers/include/controllers/impedance/Impedance.hpp [hunk 2]
```diff
@@ -78,12 +80,14 @@ Impedance<S>::Impedance(unsigned int dimensions) :
     state_representation::make_shared_parameter<Eigen::MatrixXd>(
         "damping", Eigen::MatrixXd::Identity(dimensions, dimensions))), inertia_(
     state_representation::make_shared_parameter<Eigen::MatrixXd>(
-        "inertia", Eigen::MatrixXd::Identity(dimensions, dimensions))), force_limit_(
+        "inertia", Eigen::MatrixXd::Identity(dimensions, dimensions))), forward_force_(
+            state_representation::make_shared_parameter<bool>("forward_force", true)), force_limit_(
```
Comment (eeberhard) [inline source/controllers/include/controllers/impedance/Impedance.hpp:84]: I would prefer this to be false by default

File: source/controllers/src/impedance/CompliantTwist.cpp [hunk 3]
```diff
@@ -23,6 +23,11 @@ CompliantTwist::CompliantTwist(
 
   set_linear_gains(linear_principle_damping, linear_orthogonal_damping);
   set_angular_gains(angular_stiffness, angular_damping);
+  // dissipative already doesn't have stiffness and inertia
+  // by setting forward force to false, we ensure that only the velocity error contributes to the command
+  // FIXME: is this just for the compliant twist controller or for the dissipative in general?
+  // FIXME: or should this even be the user's responsibility to set that?
+  this->dissipative_ctrl_.set_parameter_value("forward_force", false);
```
Comment (eeberhard) [inline source/controllers/src/impedance/CompliantTwist.cpp:30]: I would remove this and make it the user's choice in every case.

**Comment by domire8** [comment]
Thanks, I agree with all your comments.

**Review by eeberhard — APPROVED** [review]

**Pull request merged**